src/Stoa.ts

```diff
@@ -21,11 +23,41 @@ class Stoa {
             res.send(req.params.address + ':' + req.query.height);
         });
 
+        /**
+         * When a request is received through agora-history push
+         * JSON block data is parsed and stored on each storage.
+         */
         this.stoa.post("/push",
-            (req: express.Request, res: express.Response, next: express.NextFunction) => {
-            console.log(req.body);
-            res.status(200).send();
-        });
+            asyncHandler(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            var data: any = req.body;
+            var block: any;
+
+            if (data.block == undefined)
+                res.status(500).send("Block data is not defined");
```

Comment:
what happens after the if statement is true? We send back 500, and then? Does the delegate return? Or is the code missing return statements?

Reply:
Yes, I further subdivided the error-treatment logic.